File: facade/java/CreditCardFacade.java
package Facade;

// Banka müşterimizi temsili oluşturan sınıf.
public class CreditCardFacade {
    public void createCreditCart(Customer customer) {
        CreditCardManager creditCardManager = new CreditCardManager();

        BlackListService blackListService = new BlackListService();

        if (!blackListService.checkEmployeeIsTheBlackList(customer)) {
            creditCardManager.createCreditCart(customer);
        }
    }
}

File: facade/kotlin/CreditCardFacade.kt
package Facade

// Banka müşterimizi temsili oluşturan sınıf.
class CreditCardFacade {
    fun createCreditCart(customer: Customer) {
        val creditCardManager = CreditCardManager()

        val blackListService = BlackListService()

        if (!blackListService.checkEmployeeIsTheBlackList(customer)) {
            creditCardManager.createCreditCart(customer)
        }
    }
}
